interface/modules/custom_modules/oe-module-faxsms/src/Controller/AppDispatch.php

```diff
@@ -0,0 +1,439 @@
+                    call_user_func(array($this, $action), array())
+                );
+            } else {
+                $this->setHeader("HTTP/1.0 404 Not Found");
+                die(xlt("Requested") . ' ' . text($action) . ' ' . xlt("or service is not found.") . '<br />' . xlt("Install or turn service on!"));
+            }
+        } else {
+            // Not an internal route so pass on to current service index action..
+            $this->setResponse(
+                call_user_func(array($this, self::ACTION_DEFAULT), array())
+            );
+        }
+    }
+
+    /**
+     * @param $param
+     * @param $default
+     * @return mixed|null
+     */
+    public function getSession($param = null, $default = null): mixed
+    {
+        if ($param) {
+            return $_SESSION[$param] ?? $default;
+        }
+
+        return $this->_session;
+    }
+
+    /**
+     * @param $param
+     * @param $default
+     * @return mixed|null
+     */
+    public function getQuery($param = null, $default = null): mixed
+    {
+        if ($param) {
+            return $this->_query[$param] ?? $default;
+        }
+
+        return $this->_query;
+    }
+
+    /**
+     * @param $content
+     * @return void
+     */
+    public function setResponse($content): void
+    {
+        $this->_response = $content;
+    }
+
+    /**
+     * @param $params
+     * @return $this
+     */
+    public function setHeader($params): static
+    {
+        if (!headers_sent()) {
+            if (is_scalar($params)) {
+                header($params);
+            } else {
+                foreach ($params as $key => $value) {
+                    header(sprintf('%s: %s', $key, $value));
+                }
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * @return void
+     * @throws \Exception
+     */
+    private function render(): void
+    {
+        if ($this->_response) {
+            if (is_scalar($this->_response)) {
+                echo $this->_response;
+            } else {
+                throw new \Exception(xlt('Response content must be scalar'));
+            }
+
+            exit;
+        }
+    }
+
+    /**
+     * This is where we decide which Api to use.
+     *
+     * @param string $type
+     * @return EtherFaxActions|RCFaxClient|TwilioSMSClient|void|null
+     */
+    static function getApiService(string $type)
+    {
+        try {
+            if (empty($type)) {
+                $type = $_REQUEST['type'] ?? $_SESSION["current_module_type"] ?? null;
+            }
+            self::setModuleType($type);
+            self::$_apiService = self::getServiceInstance($type);
+            return self::$_apiService;
+        } catch (\Exception $e) {
+            echo $e->getMessage();
+            exit;
+        }
+    }
+
+    /**
+     * @param $type
+     * @return void
+     */
+    static function setModuleType($type): void
+    {
+        $_SESSION['current_module_type'] = $type;
```

Comment:
I'd recommend prefixing this with something like oefax_current_module_type as this value could be reused in other parts of core and its generic enough that someone could stomp you.